### PR TITLE
Support multiple instances of secnetperf in kernel mode

### PR DESCRIPTION
**Pull request by ThadHouse**
This way loopback tests can be supported. Done by allowing user to supply driver name

### CODE REVIEW

File: src/perf/bin/drvmain.cpp [hunk 1]
```diff
@@ -219,11 +215,45 @@ SecNetPerfDriverUnload(
...
+        ServiceNameLength++;
+    }
+    if (ServiceNameLength == 0) {
+        //LogError("[config] Empty service name!");
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    UNICODE_STRING ServiceString = { ServiceNameLength, ServiceNameLength, BaseRegPath->Buffer + (BaseRegPathLength - ServiceNameLength) };
+
+    return RtlUnicodeStringCopy(ServiceName, &ServiceString);
```
Comment (nibanks) [inline src/perf/bin/drvmain.cpp:248]: Do we really need to copy, or could we just reference the original memory?

Comment (ThadHouse) [reply src/perf/bin/drvmain.cpp:248]: For this, yeah I could probably just reference the original memory.

File: src/perf/bin/appmain.cpp [hunk 2]
```diff
@@ -295,7 +290,7 @@ QuicKernelMain(
             OutBuffer,
             OutBufferSize,
             &OutBufferWritten,
-            240000);
+            INFINITE);
```
Comment (anrossi) [inline src/perf/bin/appmain.cpp:293]: is this safe? Seems like a fun way to have a new hang

Comment (ThadHouse) [reply src/perf/bin/appmain.cpp:293]: Yes, this is safe to do. This is basically the wait for test to finish call. Which in server mode can literally be forever, as we just want it to stay running. Cancelling with ctrl c, the shutdown backdoor, and a client test actually finishing will cause this to return.

File: src/perf/lib/SecNetPerfMain.cpp [hunk 3]
```diff
@@ -139,6 +139,11 @@ QuicMainStart(
         if (QUIC_FAILED(Status)) {
             CxPlatDataPathUninitialize(Datapath);
             Datapath = nullptr;
+            //
+            // Must explicitly set binding to null, as CxPlatSocketCreateUdp
+            // can set the Binding variable even in invalid cases.
```
Comment (nibanks) [inline src/perf/lib/SecNetPerfMain.cpp:144]: Well, it should clear it then. Can you fix that with a separate PR?

**Review by nibanks — APPROVED** [review]

File: src/perf/lib/SecNetPerfMain.cpp (hunk 3 above)
Comment (ThadHouse) [reply src/perf/lib/SecNetPerfMain.cpp:144]: Yes, I can do that in a separate PR.

**Pull request merged**